apps/api/pages/api/users/[userId]/_patch.ts

```diff
@@ -94,9 +97,10 @@ export async function patchHandler(req: NextApiRequest) {
   if (!isAdmin && query.userId !== req.userId) throw new HttpError({ statusCode: 403, message: "Forbidden" });
 
   const body = await schemaUserEditBodyParams.parseAsync(req.body);
-  // disable role changes unless admin.
-  if (!isAdmin && body.role) {
-    body.role = undefined;
+  // disable role or branding changes unless admin.
+  if (!isAdmin) {
+    if (body.role) body.role = undefined;
+    if (body.hideBranding) body.hideBranding = undefined;
```

Comment:
Am I incorrect in thinking this is going to actually override the hideBranding value that's already stored in the database instead of just ignoring what's on the body since we just use the full body object on line 199?

Reply:
This would ignore "hideBranding" as the only acceptable values for hideBranding are boolean, and undefined is pretty much as if it hasn't been passed (since this is optional) in validation schema.

Reply:
Cool